### PR TITLE
Add support for accessing ListView/TreeView itemDelegate

### PR DESCRIPTION
**Pull request by Znurre**
This commit adds support for getting/setting itemDelegate for ListView and TreeView

### CODE REVIEW

File: src/declarativetreeviewextension_p.h [hunk 1]
```diff
@@ -45,22 +46,27 @@ class DeclarativeTreeViewExtension : public DeclarativeItemViewExtension
...
   Q_PROPERTY(QItemSelectionModel* selectionModel READ selectionModel WRITE setSelectionModel NOTIFY selectionModelChanged)
+  Q_PROPERTY(QAbstractItemDelegate* itemDelegate READ itemDelegate WRITE setItemDelegate NOTIFY itemDelegateChanged)
 
   Q_PROPERTY(QQmlListProperty<QObject> data READ data DESIGNABLE false)
 
   Q_CLASSINFO("DefaultProperty", "data")
 
   public:
-    explicit DeclarativeTreeViewExtension(QObject *parent = 0);
+    explicit DeclarativeTreeViewExtension(QObject *parent = Q_NULLPTR);
```
Comment (krake-kdab) [inline src/declarativetreeviewextension_p.h]: In general a valid change, but not related to support for delegates.
Such a cleanup should be in a separate commit and probably across classes.

File: src/declarativewidgets_plugin.cpp [hunk 2]
```diff
@@ -101,6 +101,7 @@
 #include <QToolBar>
 #include <QToolButton>
 #include <QTreeView>
+#include <QAbstractItemDelegate>
```
Comment (krake-kdab) [inline src/declarativewidgets_plugin.cpp]: I think includes are usually in alphabetical order

File: src/declarativewidgets_plugin.cpp [hunk 3]
```diff
@@ -115,6 +116,7 @@ void ExtensionpluginPlugin::registerTypes(const char *uri)
   // uncreatable core
   qmlRegisterType<QAbstractItemModel>();
   qmlRegisterType<QItemSelectionModel>();
+  qmlRegisterType<QAbstractItemDelegate>();
```
Comment (krake-kdab) [inline src/declarativewidgets_plugin.cpp]: QAbstractItemDelegate can not be constructed in C++, since it is abstract.
So this would need to be registered with qmlRegisterUncreatableType()

Comment (Znurre) [reply src/declarativewidgets_plugin.cpp]: I beg to differ. It's using the qmlRegisterType overload without parameters, only to make the type known to QML.
Same thing can be seen 2 lines above for QAbstractItemModel.
Should this really be handled with qmlRegisterUncreatableType?

**Comment by Znurre** [comment]
My use case for this is simply instantiating a custom delegate in QML and assigning it to a ListView/TreeView as delegate.

File: src/declarativewidgets_plugin.cpp (hunk 3 above)
Comment (krake-kdab) [reply src/declarativewidgets_plugin.cpp]: Hmm, good point.
I guess we should change that to uncreatable type registration, in a future clean up commit, and for the other uncreatables as well

**Comment by krake-kdab** [comment]
> My use case for this is simply instantiating a custom delegate in QML and assigning it to a ListView/TreeView as delegate

A cool. We should probably add an example like that :-)
